flow/connectors/mysql/mysql.go

```diff
@@ -1,15 +1,125 @@
-// stub to bypass validation
+package connmysql
 
-package mysql
+import (
+	"context"
+	"crypto/tls"
+	"errors"
+	"fmt"
+	"iter"
+	"log/slog"
+	"time"
 
-import "context"
+	"github.com/go-mysql-org/go-mysql/client"
+	"github.com/go-mysql-org/go-mysql/mysql"
+	"go.temporal.io/sdk/log"
 
-type MySqlConnector struct{}
+	metadataStore "github.com/PeerDB-io/peerdb/flow/connectors/external_metadata"
+	"github.com/PeerDB-io/peerdb/flow/generated/protos"
+	"github.com/PeerDB-io/peerdb/flow/shared"
+)
 
-func (MySqlConnector) Close() error {
-	return nil
+type MySqlConnector struct {
+	*metadataStore.PostgresMetadata
+	config *protos.MySqlConfig
+	conn   map[context.Context]*client.Conn
+	logger log.Logger
+}
+
+func NewMySqlConnector(ctx context.Context, config *protos.MySqlConfig) (*MySqlConnector, error) {
+	pgMetadata, err := metadataStore.NewPostgresMetadata(ctx)
+	if err != nil {
+		return nil, err
+	}
+	return &MySqlConnector{
+		PostgresMetadata: pgMetadata,
+		config:           config,
+		conn:             make(map[context.Context]*client.Conn),
+		logger:           shared.LoggerFromCtx(ctx),
+	}, nil
 }
 
-func (MySqlConnector) ConnectionActive(context.Context) error {
+func (c *MySqlConnector) Close() error {
+	var errs []error
+	if c.conn != nil {
+		for _, conn := range c.conn {
+			errs = append(errs, conn.Close())
+		}
+		c.conn = nil
+	}
+	return errors.Join(errs...)
+}
+
+func (c *MySqlConnector) ConnectionActive(context.Context) error {
 	return nil
 }
+
+func (c *MySqlConnector) connect(ctx context.Context) (*client.Conn, error) {
+	conn := c.conn[ctx]
+	if conn == nil {
+		argF := []client.Option{func(conn *client.Conn) error {
+			conn.SetCapability(mysql.CLIENT_COMPRESS)
```

Comment:
https://github.com/go-mysql-org/go-mysql/blob/97aa9858c8dc7ec8c22d7f37e7dcc444fb5a3ed0/driver/driver_options.go#L36 implies ZSTD is also an option

might want to either make it configurable to use this; also I'm not sure if the server gracefully handles this

Reply:
that would fit as a followup adding compression mode to mysql config. Removed setting this capability in this PR, extending proto being kept to main mysql PR or will be split off alongside flavor